pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -1115,6 +1271,19 @@ func (s *FlinkStateMachine) isIncompatibleDeploymentModeChange(application *v1be
 	return application.Spec.DeploymentMode != application.Status.DeploymentMode
 }
 
+// Returns true if the only change between the current status and spec is an _increase_ in parallelism
+func isScaleUp(app *v1beta1.FlinkApplication) bool {
+	if app.Spec.Parallelism > app.Status.JobStatus.Parallelism {
+		appWithOldScale := app.DeepCopy()
+		appWithOldScale.Spec.Parallelism = app.Status.JobStatus.Parallelism
+		hash := flink.HashForApplication(appWithOldScale)
```

Comment:
Is it possible that the deepcopy generates a hash that's != deployHash/failedDeployHash? I seem to have run into this while testing locally. Essentially had an app w. config:
```yaml
spec:
  image: lyft/operator-test-app:b1b3cb8e8f98bd41f44f9c89f8462ce255e0d13f.1
  imagePullSecrets:
    - name: dockerhub
  flinkConfig:
    state.backend.fs.checkpointdir: file:///checkpoints/flink/checkpoints
    state.checkpoints.dir: file:///checkpoints/flink/externalized-checkpoints
    state.savepoints.dir: file:///checkpoints/flink/savepoints
  jobManagerConfig:
    offHeapMemoryFraction: 0.5
    resources:
      requests:
        memory: "100Mi"
        cpu: "0.2"
    replicas: 1
  taskManagerConfig:
    taskSlots: 2
    offHeapMemoryFraction: 0.2
    resources:
      requests:
        memory: "200Mi"
        cpu: "1"
  volumeMounts:
    - mountPath: /checkpoints
      name: checkpoints
  volumes:
    - name: checkpoints
      hostPath:
        path: /tmp/checkpoints
        type: Directory
  flinkVersion: "1.9"
  jarName: "operator-test-app-1.0.0-SNAPSHOT.jar"
  parallelism: 1
  entryClass: "com.lyft.OperatorTestApp"
  scaleMode: "InPlace"
```
Updated parallelism from 1 --> 4 but the generated hash in this method did not match the `DeployHash` (and hence the app never went into a Rescaling state).

Totally possible that it's something w. my local setup but couldn't think of anything that would affect this behaviour (FWIW I re-ran codegen)

Reply:
Did you make sure that only the parallelism changed? You can't use spcli for this currently, as it sets annotations that change the hash.

I think the easiest way to test is to use `kubectl edit` directly.

Reply:
I was using `kubectl apply`. `kubectl edit` seems to work correctly, though unclear why that would be the case b/c I assumed underneath both commands use PATCH to update resources.

Reply:
Kiubectl apply actually modifies the annotations as well, in order to track which fields it controls.

Reply:
I guess expanding on this a bit, it unfortunate that this feature is not compatible with changes to annotations of the FlinkApplication resource. This happens because we made the decision early on to propagate all annotations on the FlinkApplication to the underlying deployments and pods.

But the pods themselves are immutable, so if we can't do an in-place deploy that involves changes to annotations. If we wanted to support this correctly, we'd have to separate out the annotations on the FlinkApplication resource from the annotations that we set on the deployments and pods.

Reply:
The difference in behavior of `kubectl apply` makes sense now. Thanks!

I agree that this is a bit non-ideal. It maybe worth adding a note about the incompatibility to docs/release notes. Also, I realized that with this change, pods can no longer be queried using the `flink-app-hash` label? I don't think this would have been pervasively used, but also maybe useful to document.

Reply:
Added a note about this to the CRD documentation. Longer-term (especially if we're thinking about releasing v1 of the CRD) we may want to look at changing this decision around labels and annotations

`flink-app-hash` is now an annotation on the pods, rather than a label, so it can't be queried in the same way. It's also not going to be consistent across pods in the case of an in-place deploy. I've always considered the app hashes an implementation detail of the operator, and not something that users should be relying on. I don't think we document the current behavior anywhere, so I think this is ok to change.